File: rb/features.rb
# Numbers SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module NumbersFeatures
  def self.make_feature(name)
    case name
    when "base"
      NumbersBaseFeature.new
    when "ratelimit"
      NumbersRatelimitFeature.new
    when "retry"
      NumbersRetryFeature.new
    when "test"
      NumbersTestFeature.new
    when "timeout"
      NumbersTimeoutFeature.new
    else
      NumbersBaseFeature.new
    end
  end
end

File: py/features.py
# Numbers SDK feature factory

from feature.base_feature import NumbersBaseFeature
from feature.test_feature import NumbersTestFeature


def _make_feature(name):
    features = {
        "base": lambda: NumbersBaseFeature(),
        "test": lambda: NumbersTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
